lib/livebook_web/live/settings_live.ex

```diff
@@ -286,16 +335,32 @@ defmodule LivebookWeb.SettingsLive do
     handle_event("set_autosave_path", %{}, socket)
   end
 
+  def handle_info({:env_vars_changed, env_vars}, socket) do
+    {:noreply, assign(socket, env_vars: env_vars)}
+  end
+
   def handle_info(_message, socket), do: {:noreply, socket}
 
   defp autosave_dir() do
-    Livebook.Settings.autosave_path()
+    path =
+      case Livebook.Settings.autosave_path() do
+        nil -> System.tmp_dir!()
+        path -> path
+      end
+
+    path
     |> Livebook.FileSystem.Utils.ensure_dir_path()
     |> Livebook.FileSystem.File.local()
```

Comment:
@aleDsz I mean this :)

```suggestion
    if path = Livebook.Settings.autosave_path() do
      path
      |> Livebook.FileSystem.Utils.ensure_dir_path()
      |> Livebook.FileSystem.File.local()
    end
```